src/commands/synthetics/api.ts

```diff
@@ -160,39 +160,85 @@ const getTunnelPresignedURL = (request: (args: AxiosRequestConfig) => AxiosPromi
   return resp.data
 }
 
-const getMobileApplicationPresignedURL = (
-  request: (args: AxiosRequestConfig) => AxiosPromise<PresignedUrlResponse>
-) => async (applicationId: string, appSize: number, md5: string) => {
+const getMobileApplicationPresignedURLs = (
+  request: (args: AxiosRequestConfig) => AxiosPromise<MultipartPresignedUrlsResponse>
+) => async (
+  applicationId: string,
+  appSize: number,
+  parts: MobileApplicationUploadPart[]
+): Promise<MultipartPresignedUrlsResponse> => {
+  const partForRequest = (part: MobileApplicationUploadPart) => ({
+    md5: part.md5,
+    partNumber: part.partNumber,
+  })
+
   const resp = await retryRequest(
     {
-      data: {appSize, md5},
+      data: {
+        appSize,
+        parts: parts.map(partForRequest),
+      },
       method: 'POST',
-      url: `/synthetics/mobile/applications/${applicationId}/presigned-url`,
+      url: `/synthetics/mobile/applications/${applicationId}/multipart-presigned-urls`,
     },
     request
   )
 
   return resp.data
 }
 
-const uploadMobileApplication = (request: (args: AxiosRequestConfig) => AxiosPromise<void>) => async (
-  fileBuffer: Buffer,
-  presignedUrlParams: PresignedUrlResponse['presigned_url_params']
-) => {
-  const form = new FormData()
-  Object.entries(presignedUrlParams.fields).forEach(([key, value]) => {
-    form.append(key, value)
+const uploadMobileApplicationPart = (request: (args: AxiosRequestConfig) => AxiosPromise<void>) => async (
+  parts: MobileApplicationUploadPart[],
+  multipartPresignedUrlsParams: MultipartPresignedUrlsResponse['multipart_presigned_urls_params']
+): Promise<MobileApplicationUploadPartResponse[]> => {
+  const promises = Object.entries(multipartPresignedUrlsParams.urls).map(async ([partNumber, presignedUrl]) => {
+    const resp = await retryRequest(
+      {
+        data: parts[Number(partNumber) - 1].blob,
+        headers: {
+          'Content-MD5': parts[Number(partNumber) - 1].md5,
+          // Presigned URL *requires* unset content-type since it's used for signature
+          // We can clear axios default by setting to null
+          // https://github.com/axios/axios/issues/382
```

Comment:
**nit**: this link is the PR that closes the issue, and it explicitly talks about `null` (for the curious who would try to use `undefined` instead)
```suggestion
          // https://github.com/axios/axios/pull/1845
```

Reply:
Oh ya -- good suggestion 👍